internal/hardware/validator.go

```diff
@@ -173,6 +175,27 @@ func (v *validator) GetClusterHostRequirements(ctx context.Context, cluster *com
 	}, nil
 }
 
+func (v *validator) GetInfraEnvHostRequirements(ctx context.Context, infraEnv *common.InfraEnv, host *models.Host) (*models.ClusterHostRequirements, error) {
```

Comment:
Is it intentional that 'GetInfraEnvHostRequirements' returns 'ClusterHostRequirements'?
Shouldn't we have a dedicated structure for 'InfraEnvHostRequirements'?

Reply:
Currently it is intentional. This commit is not the final design. It should support current day1 flow and allow us to push changes in state machine and instruction manager. The final design will be done in MGMT-7366 by Ori

Reply:
Why do we need requirements for infra-env ? is it mandatory for late binding ? Do infra-env and cluster both have requirements ?

Reply:
Infra env also has requirement for hardware, but without operators requirements etc'